packages/otel/src/otel-exchange.ts

```diff
@@ -0,0 +1,74 @@
+import { type Span, SpanKind, SpanStatusCode, context } from "@opentelemetry/api";
+import { type CombinedError, type Operation, makeOperation, mapExchange } from "urql";
+import { getOtelTracer } from "./otel-tracer";
+import { AttributeNames, ObservabilityAttributes } from "./lib/observability-attributes";
+import { addInputVariableAttributes, addRequestHeaderAttributes } from "./otel-graphql-utils";
+import { SemanticAttributes } from "@opentelemetry/semantic-conventions";
+
+type Definition = {
+  name: {
+    value: string;
+  };
+};
+
+interface ExtendedFetchOptions extends RequestInit {
+  headers: Record<string, string>;
+}
+
+type ExtendedOperationContext = Operation["context"] & {
+  span: Span;
+  fetchOptions?: ExtendedFetchOptions;
+};
+
+interface ExtendedOperation extends Operation {
+  context: ExtendedOperationContext;
+}
+
+export const otelExchange = mapExchange({
```

Comment:
I still think we should only export `otelExchange` from otel package and you should construct `instrumentedGraphqlClient` in app code, not in lib

imagine that I want to add another exchange, not related to otel, and I want it to be specific for one app. This design seems broken

I think we should export raw exchange or factories and construct final client in the app itself